lib/jsonschema-tools.js

```diff
@@ -515,6 +522,19 @@ function schemaPathToInfo(schemaPath, options = {}) {
     };
 }
 
+/**
+ * Cartesian product of arrays.
+ * Taken from https://stackoverflow.com/a/43053803/555565
+ * Example:
+ *  cartesianProduct(['a','b'], ['c']) => [ [ 'a', 'c' ], [ 'b', 'c' ] ]
+ *
+ * @param  {...Array} arrays
+ * @return {Array}
+ */
+function cartesianProduct(...arrays) {
+    return arrays.reduce((a, b) => _.flatMap(a, d => b.map(e => [d, e].flat())));
```

Comment:
[offtopic] This is SO unreadable, I'm impressed :) Nice that you added a comment about where it's taken from.

Reply:
Hahah, I started read-ifying but gave up because it was taking to long.